### PR TITLE
Add generator for Subiquity Dart types

### PR DESCRIPTION
**Pull request by jpnurmi**
Imported from https://github.com/canonical/ubuntu-desktop-installer/pull/965.

TLDR: Run `make generate` to generate `types.dart` from `types.py`.

Subiquity's Python data classes on the server side must match the respective Dart data classes on the client side or else bad things will happen. It's not easy to maintain a perfect match by manually comparing hundreds of lines of Python and Dart classes side by side. This PR automates that step and ensures that the two are always in sync.

The generator script takes [`subiquity/common/types.py`](https://github.com/canonical/subiquity/blob/main/subiquity/common/types.py) as input, transforms the Python data classes to Dart with help of Python's [`ast`](https://docs.python.org/3/library/ast.html) module, and outputs [`lib/src/types.dart`](https://github.com/canonical/subiquity_client.dart/blob/main/lib/src/types.dart) with the same data classes converted to Dart and Freezed.

NOTE: The script has been battle-tested as it's been in active use since around April 2022.

### CODE REVIEW

**Review by d-loose — APPROVED** [review]
I'm definitely no expert here, but after having spent some time on it, it looks very solid to me. :+1:
Would it hurt to add the unit test to the CI?

**Comment by jpnurmi** [comment]
Thank you so much! It's probably not a beauty contest winner but hey, it does the job! :D 

I'll follow up with changes to run the generator test in the CI and also add a check to ensure that types.dart is always in sync. It becomes handy especially after we start tracking multiple subiquity branches.
